srml/elections-phragmen/src/lib.rs

```diff
@@ -0,0 +1,957 @@
+
+			T::Currency::reserve(&who, T::CandidacyBond::get())
+				.map_err(|_| "candidate has not enough funds")?;
+
+			<Candidates<T>>::mutate(|c| c.insert(index, who));
+			CandidateCount::mutate(|c| *c += 1);
+		}
+
+		/// Set the desired member count; if lower than the current count, then seats will not be up
+		/// election when they expire. If more, then a new vote will be started if one is not
+		/// already in progress.
+		#[weight = SimpleDispatchInfo::FixedOperational(10_000)]
+		fn set_desired_seats(origin, #[compact] count: u32) {
+			ensure_root(origin)?;
+			DesiredSeats::put(count);
+		}
+
+		/// Remove a particular member from the set. This is effective immediately.
+		///
+		/// Note: A tally should happen instantly (if not already in a presentation
+		/// period) to fill the seat if removal means that the desired members are not met.
+		#[weight = SimpleDispatchInfo::FixedOperational(10_000)]
+		fn remove_member(origin, who: <T::Lookup as StaticLookup>::Source) {
+			ensure_root(origin)?;
+			let who = T::Lookup::lookup(who)?;
+
+			if Self::is_candidate(&who).is_ok() {
+				let members = Self::members();
+				let new_members: Vec<T::AccountId> = members
+					.into_iter()
+					.filter(|i| *i != who)
+					.collect();
+				<Members<T>>::put(new_members.clone());
+				T::Currency::unreserve(&who, T::CandidacyBond::get());
+				T::ChangeMembers::change_members(&[], &[who], new_members);
+			}
+		}
+
+		/// Set the presentation duration. If there is current a vote being presented for, will
+		/// invoke `finalize_vote`.
+		#[weight = SimpleDispatchInfo::FixedOperational(10_000)]
+		fn set_term_duration(origin, #[compact] count: T::BlockNumber) {
+			ensure_root(origin)?;
+			<TermDuration<T>>::put(count);
+		}
+
+		fn on_initialize(n: T::BlockNumber) {
+			if let Err(e) = Self::end_block(n) {
+				runtime_io::print("Guru meditation");
+				runtime_io::print(e);
+			}
+		}
+	}
+}
+
+decl_event!(
+	pub enum Event<T> where <T as system::Trait>::AccountId {
+		/// A new term with new members.
+		NewTerm(Vec<AccountId>),
+		/// No candidates were elected for this round.
+		EmptyCouncil(),
+	}
+);
+
+impl<T: Trait> Module<T> {
+	/// Check if `who` is a candidate. It returns the insert index if the element does not exists as
+	/// an error.
+	fn is_candidate(who: &T::AccountId) -> Result<(), usize> {
+		Self::candidates().binary_search(who).map(|_| ())
+	}
+
+	/// Check if `who` is a voter in the ongoing round.
+	fn is_current_voter(who: &T::AccountId) -> bool {
+		let now = Self::election_rounds();
+		Self::votes_of(who).1 == now && <StakeOf<T>>::exists(who)
+	}
+
+	/// Check if `who` was a voter who has not yet called `remove_voter`.
+	fn is_old_voter(who: &T::AccountId) -> bool {
+		let now = Self::election_rounds();
+		Self::votes_of(who).1 < now && <StakeOf<T>>::exists(who)
+	}
+
+	/// The locked stake of a voter.
+	fn locked_stake_of(who: &T::AccountId) -> BalanceOf<T> {
+		if Self::is_current_voter(who) {
```

Comment:
This is super-expensive for a sanity check. Why not just use `stake_of` directly?

Reply:
yeah -- my msitake.